azurerm/internal/services/compute/tests/resource_arm_dedicated_host_test.go

```diff
@@ -0,0 +1,328 @@
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				// Enabled
+				Config: testAccAzureRMDedicatedHost_autoReplaceOnFailure(data, true),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDedicatedHost_licenseType(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dedicated_host", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDedicatedHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDedicatedHost_licenceType(data, "None"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMDedicatedHost_licenceType(data, "Windows_Server_Hybrid"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMDedicatedHost_licenceType(data, "Windows_Server_Perpetual"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMDedicatedHost_licenceType(data, "None"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDedicatedHost_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dedicated_host", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDedicatedHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDedicatedHost_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDedicatedHost_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dedicated_host", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDedicatedHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDedicatedHost_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMDedicatedHost_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDedicatedHostExists(data.ResourceName),
```

Comment:
Worth checking the properties are actually updated to expected values here?

Reply:
the import'll handle that - so this should be fine 👍